skrf/vi/validators.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any, Sequence
+
+import re
+from abc import ABC, abstractmethod
+from enum import Enum
+
+
+class ValidationError(Exception):
+    pass
+
+
+class Validator(ABC):
+    @abstractmethod
+    def validate_input(self, arg) -> Any:
+        """Try to convert arg to a valid value and return. Used to ensure
+        commands sent to the instrument are the proper format."""
+        pass
+
+    def validate_output(self, arg) -> Any:
+        """Try to convert arg to a valid value and return. Used to convert
+        responses from the instrument to something useful. By default,
+        this just calls validate_input, but can be overwritten in subclasses.
+        (See EnumValidator)"""
+        return self.validate_input(arg)
+
+
+class IntValidator(Validator):
+    def __init__(self, min: int | None = None, max: int | None = None) -> None:
```

Comment:
such syntax will fail with Python <3.10. Eventually please use the more verbose Union[int, None] for the moment, as we follows the Python end of life dates https://endoflife.date/python

Reply:
I noticed this in the gh action. I'll fix with the next commit.